api/types/src/transaction.rs

```diff
@@ -1087,25 +1091,98 @@ impl TryFrom<MultiEd25519Signature> for AccountAuthenticator {
+    pub signature: HexEncodedBytes,
+}
+
+impl VerifyInput for Secp256k1Signature {
+    fn verify(&self) -> anyhow::Result<()> {
+        let public_key_len = self.public_key.inner().len();
+        let signature_len = self.signature.inner().len();
+        if public_key_len != secp256k1::PUBLIC_KEY_LENGTH {
+            bail!(
+                "Secp256k1 signature's public key is an invalid number of bytes, should be {} bytes but found {}",
+                secp256k1::PUBLIC_KEY_LENGTH, public_key_len
+            )
+        } else if signature_len != secp256k1::SIGNATURE_LENGTH {
+            bail!(
+                "Secp256k1 signature length is an invalid number of bytes, should be {} bytes but found {}",
+                secp256k1::SIGNATURE_LENGTH, signature_len
+            )
+        } else {
+            // TODO: Check if they match / parse correctly?
+            Ok(())
+        }
+    }
+}
+
+impl TryFrom<Secp256k1Signature> for TransactionAuthenticator {
+    type Error = anyhow::Error;
+
+    fn try_from(value: Secp256k1Signature) -> Result<Self, Self::Error> {
+        let Secp256k1Signature {
+            public_key,
+            signature,
+        } = value;
+        Ok(TransactionAuthenticator::secp256k1(
+            public_key
+                .inner()
+                .try_into()
+                .context("Failed to parse given public_key bytes as a Secp256k1PublicKey")?,
+            signature
+                .inner()
+                .try_into()
+                .context("Failed to parse given signature as a Secp256k1Signature")?,
+        ))
+    }
+}
+
+impl TryFrom<Secp256k1Signature> for AccountAuthenticator {
+    type Error = anyhow::Error;
+
+    fn try_from(value: Secp256k1Signature) -> Result<Self, Self::Error> {
+        let Secp256k1Signature {
+            public_key,
+            signature,
+        } = value;
+        Ok(AccountAuthenticator::secp256k1(
+            public_key
+                .inner()
+                .try_into()
+                .context("Failed to parse given public_key bytes as a Secp256k1PublicKey")?,
+            signature
+                .inner()
+                .try_into()
+                .context("Failed to parse given signature as a Secp256k1Signature")?,
+        ))
+    }
+}
+
 /// Account signature scheme
 ///
 /// The account signature scheme allows you to have two types of accounts:
 ///
 ///   1. A single Ed25519 key account, one private key
 ///   2. A k-of-n multi-Ed25519 key account, multiple private keys, such that k-of-n must sign a transaction.
+///   3. A single Secp256k1 key account, one private key
 #[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, Union)]
 #[serde(tag = "type", rename_all = "snake_case")]
 #[oai(one_of, discriminator_name = "type", rename_all = "snake_case")]
 pub enum AccountSignature {
     Ed25519Signature(Ed25519Signature),
     MultiEd25519Signature(MultiEd25519Signature),
+    Secp256k1Signature(Secp256k1Signature),
```

Comment:
Makes sense to add here. The only part that is slightly strange is when a multied25519 account can be authenticated by Secp256k1 keys. Would there be any possible problems with this?

Reply:
No, multied25519, as it is, cannot. we need a new general authenticator that supports arbitrary public keys, which the current one does not.

Reply:
also note, this is just the annoying json representation stuff... the core code is in types/src/transaction/authenticator.rs

Reply:
Hmmm then fee payer would not work with this either?

Reply:
https://github.com/aptos-labs/aptos-core/blob/main/types/src/transaction/authenticator.rs#L61

note, that the feepayer takes in AccountAuthenticator - which can be the Secp256k1
the multied25519 takes in a ... weird multied25519public key

I think the right approach would have been effectively making the multi-keys that use many account authenticators instead of shoving this into the crypto code...

Reply:
Makes sense